ledger-core/v2/logicrunner/s_messenger/messenger.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/insolar/blob/master/LICENSE.md.
+
+package s_messenger
+
+import (
+	"context"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/network/facade"
+)
+
+type MessengerService interface {
+	facade.Messenger
+}
+
+type MessengerServiceAdapter struct {
+	svc  MessengerService
+	exec smachine.ExecutionAdapter
+}
+
+func (a *MessengerServiceAdapter) PrepareSync(
+	ctx smachine.ExecutionContext,
+	fn func(svc MessengerService),
+) smachine.SyncCallRequester {
+	return a.exec.PrepareSync(ctx, func(interface{}) smachine.AsyncResultFunc {
+		fn(a.svc)
+		return nil
+	})
+}
+
+func (a *MessengerServiceAdapter) PrepareAsync(
+	ctx smachine.ExecutionContext,
+	fn func(svc MessengerService) smachine.AsyncResultFunc,
+) smachine.AsyncCallRequester {
+	return a.exec.PrepareAsync(ctx, func(interface{}) smachine.AsyncResultFunc {
+		return fn(a.svc)
+	})
+}
+
+func (a *MessengerServiceAdapter) PrepareNotify(
+	ctx smachine.ExecutionContext,
+	fn func(svc MessengerService),
+) smachine.NotifyRequester {
+	return a.exec.PrepareNotify(ctx, func(interface{}) {
+		fn(a.svc)
+	})
+}
+
+type messengerService struct {
+	facade.Messenger
+}
+
+func CreateMessengerService(messenger facade.Messenger) *MessengerServiceAdapter {
+	ctx := context.Background()
```

Comment:
propagate context please

Reply:
I did it in the same way like others are done.
But, yes, why not